api/src/main/java/com/example/api/service/user/BadgeService.java

```diff
@@ -55,10 +65,48 @@ public void checkAllBadges() throws WrongUserTypeException, EntityNotFoundExcept
                 .toList();
         for (Badge badge: badges) {
             if (badge.isGranted(badgeVisitor)) {
-                UnlockedBadge unlockedBadge = new UnlockedBadge(null, badge, System.currentTimeMillis());
+                UnlockedBadge unlockedBadge = new UnlockedBadge(badge, System.currentTimeMillis(), student);
                 unlockedBadgeRepo.save(unlockedBadge);
                 student.getUnlockedBadges().add(unlockedBadge);
             }
         }
     }
+
+    public void deleteBadge(Long badgeId) {
+        badgeRepo.deleteById(badgeId);
+    }
+
+    public void updateBadge(BadgeUpdateForm form) throws RequestValidationException, IOException {
+        Long id = form.getId();
+        Badge badge = badgeRepo.findBadgeById(id);
+        badgeValidator.validateBadgeIsNotNull(badge, id);
+
+        badge.update(form, badgeValidator);
+        badgeRepo.save(badge);
+    }
```

Comment:
Może głupie pytanie, ale czy po updateBadge sprawdzane jest czy student powinien stracić wcześniej uzyskaną odznakę, albo jakiś inny student otrzymać bo po updacie łapie się wymaganiach? Po wykonaniu jakiejś aktywności sprawdza się to checkAllBadges(), ale nie zauważyłem użycia tego w ramach updateBadge()

Reply:
no w sumie, nie pomyslalem o tym

Reply:
a nie dobra - updateBadge robi prowadzacy, a checkAllBadges sprawdza dla danego uzytkownika

Reply:
wiec bez sensu zeby mu sprawdzalo, mysle ze mozna zostawic, jak badgew sie zmieni i student bedzie spelnial jego wymagania to pzy nastepnym dodaniu punktow odblokuje mu sie odznaka

albo jak teraz mysle mozna dodac checkAllBadges na endpoint do wczytywaniua odznak wtedy jakby sobie chcial sprawdzic odznaki to nigduy nie bedzie sytuacji ze patrzy na odznaki i nie ma jakiejs ktorej wymaganie spelnil bo sie sprawdza przed wczytaniem odznak

Reply:
ale to tez sa takie casy ze mysle zeby sobie nie zawracac juz nimi glowy

Reply:
wpierdole to checkowanie badge do /all, ale to juz np jak bedzie case ze prowadzacu zmieni odznake i student nie ebdzie jej spelanil to imo zostawic studentowi i elo, prawo nie dziala wstecz, a kazdemu kolejnemuy bedzie sprawdzana


amen

Reply:
no spoko